app/src/main/java/com/islamey/hamza/wazaker/ui/AzkarListFragment/AzkarListFragment.kt

```diff
@@ -29,25 +28,21 @@ class AzkarListFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        setAdapter()
+        setUpAzkarListRecyclerView()
     }
 
-    private fun setAdapter() {
-        val layoutManager = LinearLayoutManager(context)
-        binding.AzkarListRecycler.layoutManager = layoutManager
+    private fun setUpAzkarListRecyclerView() {
 
-        val azkarListAdapter =
-            AzkarListAdapter(AzkarListDataset.getAzkarList(requireContext()))
+        val adapter = AzkarListAdapter(this)
 
-        binding.AzkarListRecycler.adapter = azkarListAdapter
+        binding.azkarListRecyclerView.adapter = adapter
+        adapter.differ.submitList(AzkarListDataset.getAzkarList(requireContext()))
 
-        azkarListAdapter.setListner(object : OnItemListener {
-            override fun onItemCLicked(item: Any?) {
+    }
 
-                val action =
-                    AzkarListFragmentDirections.actionAzkarFragmentToZekerAndDoaaFragment(item as Int)
-                findNavController().navigate(action)
-            }
-        })
+    override fun onItemCLicked(item: Any?) {
```

Comment:
please do not have `Any`

Reply:
why?

Reply:
it is not safe, we should have a type explicitly known.

Reply:
> it is not safe, we should have a type explicitly known.

it's right to create many onitemListener interfaces ?

Reply:
we can make it using generics.